DSCResources/MSFT_xWebsite/MSFT_xWebsite.psm1

```diff
@@ -576,67 +596,374 @@ function ConvertTo-CimBinding
+                    {
+                        if ($Binding.IPAddress -or $Binding.Port -or $Binding.HostName)
+                        {
+                            Write-Verbose -Message ("BindingInformation is ignored for bindings of type '$($Binding.Protocol)'" +
+                                " in case at least one of the following properties is specified: IPAddress, Port, HostName.")
+
+                            $IsJoinRequired = $true
+                        }
+                        else
+                        {
+                            $IsJoinRequired = $false
+                        }
+                    }
+                    else
+                    {
+                        $IsJoinRequired = $true   
+                    }
+
+                    # Construct the bindingInformation attribute
+                    if ($IsJoinRequired -eq $true)
+                    {
+                        try
+                        {
+                            $IPAddressString = Format-IPAddressString -InputString $Binding.IPAddress -ErrorAction Stop
+                        }
+                        catch
+                        {
+                            throw "Could not validate IP address '$($Binding.IPAddress)': '$($_.Exception.Message)'."
+                        }
+
+                        if (-not $Binding.Port)
+                        {
+                            switch ($Binding.Protocol)
+                            {
+                                'http'  {$PortNumberString = '80'}
+                                'https' {$PortNumberString = '443'}
+                            }
+
+                            Write-Verbose -Message "Port is not specified. The default '$($Binding.Protocol)' port '$PortNumberString' will be used."
+                        }
+                        else
+                        {
+                            if (Test-PortNumber -InputString $Binding.Port)
+                            {
+                                $PortNumberString = $Binding.Port
+                            }
+                            else
+                            {
+                                throw 'The port number must be a positive integer between 1 and 65535.'
+                            }
+                        }
+
+                        $BindingInformation = $IPAddressString, $PortNumberString, $Binding.HostName -join ':'
+                        $OutputObject.Add('bindingInformation', [String]$BindingInformation)
+                    }
+                    else
+                    {
+                        $OutputObject.Add('bindingInformation', [String]$Binding.BindingInformation)
+                    }
+                }
+                else
+                {
+                    if (-not $Binding.BindingInformation)
+                    {
+                        throw "BindingInformation is required for bindings of type '$($Binding.Protocol)'."
```

Comment:
Anytime we throw we should probably match the construction of our error handling that we do else where (see [line 329](https://github.com/PowerShell/xWebAdministration/pull/69/files#diff-adf4d9959eb817562f083c6961202f2bR329)) . It also centralizes the strings so if we ever decide to translate them into different languages it all propagates through.